paymentsheet/src/main/java/com/stripe/android/paymentsheet/FlowControllerCompose.kt

```diff
@@ -66,33 +47,82 @@ fun rememberPaymentSheetFlowController(
+        externalPaymentMethodConfirmHandler = null,
     )
 }
 
-@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+/**
+ * Creates a [PaymentSheet.FlowController] that is remembered across compositions. Use this method if you implement any
+ * external payment methods, as specified in your [PaymentSheet.Configuration].
+ *
+ * This *must* be called unconditionally, as part of the initialization path.
+ *
+ * If you intend to create the [com.stripe.android.model.PaymentIntent] or [com.stripe.android.model.SetupIntent] on
+ * your server, include a [createIntentCallback].
+ *
+ * @param createIntentCallback If specified, called when the customer confirms the payment or setup.
+ * @param externalPaymentMethodConfirmHandler Called when a user confirms payment for an external payment method.
+ * @param paymentOptionCallback Called when the customer's desired payment method changes.
+ * @param paymentResultCallback Called when a [PaymentSheetResult] is available.
+ */
 @Composable
 fun rememberPaymentSheetFlowController(
+    createIntentCallback: CreateIntentCallback? = null,
     externalPaymentMethodConfirmHandler: ExternalPaymentMethodConfirmHandler,
-    createIntentCallback: CreateIntentCallback,
     paymentOptionCallback: PaymentOptionCallback,
     paymentResultCallback: PaymentSheetResultCallback,
 ): PaymentSheet.FlowController {
-    UpdateExternalPaymentMethodConfirmHandler(externalPaymentMethodConfirmHandler)
-
-    return rememberPaymentSheetFlowController(
-        createIntentCallback = createIntentCallback,
+    return internalRememberPaymentSheetFlowController(
         paymentOptionCallback = paymentOptionCallback,
         paymentResultCallback = paymentResultCallback,
+        createIntentCallback = createIntentCallback,
+        externalPaymentMethodConfirmHandler = externalPaymentMethodConfirmHandler
     )
 }
 
+@Composable
+private fun internalRememberPaymentSheetFlowController(
+    createIntentCallback: CreateIntentCallback?,
+    externalPaymentMethodConfirmHandler: ExternalPaymentMethodConfirmHandler?,
+    paymentOptionCallback: PaymentOptionCallback,
+    paymentResultCallback: PaymentSheetResultCallback,
+): PaymentSheet.FlowController {
+    UpdateIntentConfirmationInterceptor(createIntentCallback)
+    UpdateExternalPaymentMethodConfirmHandler(externalPaymentMethodConfirmHandler)
+
+    val viewModelStoreOwner = requireNotNull(LocalViewModelStoreOwner.current) {
+        "PaymentSheet.FlowController must be created with access to a ViewModelStoreOwner"
+    }
+
+    val activityResultRegistryOwner = requireNotNull(LocalActivityResultRegistryOwner.current) {
+        "PaymentSheet.FlowController must be created with access to a ActivityResultRegistryOwner"
+    }
+
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    val activity = rememberActivity {
+        "PaymentSheet.FlowController must be created in the context of an Activity"
+    }
+
+    return remember {
```

Comment:
This isn't new, so definitely not worth blocking on, but we should pass things into the remember (`createIntentCallback,` `externalPaymentMethodConfirmHandler`, `paymentOptionCallback`, `paymentResultCallback`).

Reply:
https://github.com/stripe/stripe-android/pull/8480